WalletWasabi.Fluent/ViewModels/Wallets/Home/History/HistoryViewModel.cs

```diff
@@ -106,7 +106,14 @@ private async Task UpdateAsync()
 					{
 						if (_transactions.FirstOrDefault(x => x.Id == newItem.Id) is { } item)
 						{
-							item.Update(newItem);
+							if (newItem is CoinJoinsHistoryItemViewModel model)
+							{
+								item.Update(model);
+							}
+							else
+							{
+								item.Update(newItem);
+							}
```

Comment:
I think this won't fix the issue. The main issue is that it tries to update a `CoinJoinsHistoryItemViewModel` with a `TransactionHistoryItemViewModel` which should not be possible. Because that means the same transaction was detected as a CJ first, then later as a normal transaction.

Reply:
Yes, The wrong `newItem` has `IsLikelyCoinJoin = false` in its `TransactionSummary`.